spaceship/lib/spaceship/portal/provisioning_profile.rb

```diff
@@ -215,31 +237,43 @@ def create!(name: nil, bundle_id: nil, certificate: nil, devices: [], mac: false
           if devices.nil? or devices.count == 0
             if self == Development or self == AdHoc
               # For Development and AdHoc we usually want all compatible devices by default
-              if mac
-                devices = Spaceship::Device.all_macs
-              else
-                devices = Spaceship::Device.all_for_profile_type(self.type)
-              end
+              devices = if platform == Spaceship::Portal::App::MAC
+                          Spaceship::Device.all_macs
+                        elsif platform == Spaceship::Portal::App::IOS and !sub_platform.nil? and sub_platform.casecmp('tvos') == 0
```

Comment:
prefer `&&` and `||` as the word modifiers sometimes have surprising precedence rules :+1: